### DISCUSSION TITLE
AutoForm formtype supporting Meteor.callAsync

### DISCUSSION THREAD

**Discussion by polygonwood** [aldeed/meteor-autoform#1736] [Ideas]
Having little experience in contributing, I'm starting this discussion. While slowly adapting my major application to Meteor 3.0, I have added some new formtypes to AutoForm to make async calls in the submit loop.
I have added this to the **branch async** of the fork [autoform-async](https://github.com/polygonwood/meteor-autoform.git) .
This works for asynchronous methods or inserts. This does not (yet) support asynchronous hooks (rare in my application).
Who can comment on this ?

**Comment by jankapunkt** [comment]
Hey @polygonwood thanks a lot for this. I have created a comparison between this repo`s `devel` branch and your `async` branch and it looks quite good so far: https://github.com/Meteor-Community-Packages/meteor-autoform/compare/devel...polygonwood:meteor-autoform:async

My suggestion is, you simply create a pull request with your `async` branch, targeting `devel`. I will mostly make sure that tests and lint run, but apart from that it should be fine.

What's missing currently: 

- insertAsync.js and methodAsync.js are not imported in dynamic.js and static.js

It's a good starting point for async migration as we can move step by step. We should keep this discussion open and add anything that will come up during async migration.
